client/state/themes/actions.js

```diff
@@ -362,7 +362,8 @@ export function activateTheme( themeId, siteId, source = 'unknown', purchased =
 
 		return wpcom.undocumented().activateTheme( themeId, siteId )
 			.then( ( theme ) => {
-				dispatch( themeActivated( theme, siteId, source, purchased ) );
+				const themeStylesheet = theme.stylesheet || themeId; // Fall back to ID for Jetpack sites which don't return a stylesheet attr.
```

Comment:
No, for Jetpack sites, sites state (`options.theme_slug`) uses the un-prefixed theme ID, too.